flink-core/src/main/java/org/apache/flink/api/common/serialization/SerializerConfigImpl.java

```diff
@@ -574,4 +575,31 @@ private void registerTypeWithTypeInfoFactory(
     public ExecutionConfig getExecutionConfig() {
         return executionConfig;
     }
+
+    @Override
+    public SerializerConfigImpl copy() {
+        final SerializerConfigImpl newSerializerConfig = new SerializerConfigImpl();
+        newSerializerConfig.configure(configuration, this.getClass().getClassLoader());
+
+        getRegisteredTypesWithKryoSerializers()
+                .forEach(
+                        (c, s) ->
+                                newSerializerConfig.registerTypeWithKryoSerializer(
+                                        c, s.getSerializer()));
+        getRegisteredTypesWithKryoSerializerClasses()
+                .forEach(newSerializerConfig::registerTypeWithKryoSerializer);
+        getDefaultKryoSerializers()
+                .forEach(
+                        (c, s) ->
+                                newSerializerConfig.addDefaultKryoSerializer(c, s.getSerializer()));
+        Optional.ofNullable(isForceKryoAvroEnabled().getAsBoolean())
```

Comment:
If I understand it correctly, it is not needed as we've already configured it in `newSerializerConfig#configure`.

Reply:
Correct, yes there's a redundancy.